src/plots/smith/set_convert.js

```diff
@@ -0,0 +1,217 @@
+        };
+
+        ax.g2c = function(v) {
+            return ax.l2c(v + rl0 - b2);
+        };
+
+        ax.g2p = function(v) { return v * m; };
+        ax.c2p = function(v) { return ax.g2p(ax.c2g(v)); };
+    };
+}
+
+function toRadians(v, unit) {
+    return unit === 'degrees' ? deg2rad(v) : v;
+}
+
+function fromRadians(v, unit) {
+    return unit === 'degrees' ? rad2deg(v) : v;
+}
+
+function setConvertAngular(ax, smithLayout) {
+    var axType = ax.type;
+
+    if(axType === 'linear') {
+        var _d2c = ax.d2c;
+        var _c2d = ax.c2d;
+
+        ax.d2c = function(v, unit) { return toRadians(_d2c(v), unit); };
+        ax.c2d = function(v, unit) { return _c2d(fromRadians(v, unit)); };
+    }
+
+    // override makeCalcdata to handle thetaunit and special theta0/dtheta logic
+    ax.makeCalcdata = function(trace, coord) {
+        var arrayIn = trace[coord];
+        var len = trace._length;
+        var arrayOut, i;
+
+        var _d2c = function(v) { return ax.d2c(v, trace.thetaunit); };
+
+        if(arrayIn) {
+            if(Lib.isTypedArray(arrayIn) && axType === 'linear') {
+                if(len === arrayIn.length) {
+                    return arrayIn;
+                } else if(arrayIn.subarray) {
+                    return arrayIn.subarray(0, len);
+                }
+            }
+
+            arrayOut = new Array(len);
+            for(i = 0; i < len; i++) {
+                arrayOut[i] = _d2c(arrayIn[i]);
+            }
+        } else {
+            var coord0 = coord + '0';
+            var dcoord = 'd' + coord;
+            var v0 = (coord0 in trace) ? _d2c(trace[coord0]) : 0;
+            var dv = (trace[dcoord]) ? _d2c(trace[dcoord]) : (ax.period || 2 * Math.PI) / len;
+
+            arrayOut = new Array(len);
+            for(i = 0; i < len; i++) {
+                arrayOut[i] = v0 + i * dv;
+            }
+        }
+
+        return arrayOut;
+    };
+
+    // N.B. we mock the axis 'range' here
+    ax.setGeometry = function() {
+        var sector = smithLayout.sector;
+        var sectorInRad = sector.map(deg2rad);
+        var dir = {clockwise: -1, counterclockwise: 1}[ax.direction];
+        var rot = deg2rad(ax.rotation);
+
+        var rad2g = function(v) { return dir * v + rot; };
+        var g2rad = function(v) { return (v - rot) / dir; };
+
+        var rad2c, c2rad;
+        var rad2t, t2rad;
+
+        switch(axType) {
+            case 'linear':
+                c2rad = rad2c = Lib.identity;
+                t2rad = deg2rad;
+                rad2t = rad2deg;
+
+                // Set the angular range in degrees to make auto-tick computation cleaner,
+                // changing rotation/direction should not affect the angular tick value.
+                ax.range = Lib.isFullCircle(sectorInRad) ?
+                    [sector[0], sector[0] + 360] :
+                    sectorInRad.map(g2rad).map(rad2deg);
+                break;
+
+            case 'category':
```

Comment:
Should *"category"* `type` be supported here?

Reply:
Nope, removed in e17fcef3922bd77cb7fe28698a4b93538a0490c9.